tests/test_layabout.py

```diff
@@ -1 +1,327 @@
-import layabout  # noqa: F401
+import os
+from unittest.mock import MagicMock, call
+
+import pytest
+
+from layabout import (
+    Layabout,
+    MissingSlackToken,
+    FailedConnection,
+)
+
+# The likelihood of this ever being a valid Slack token is probably slim.
+TOKEN = 'xoxb-13376661337-DeAdb33Fd15EA53fACef33D1'
+
+
+@pytest.fixture
+def slack_client():
+    def api_call(method):
+        return dict(
+            channels=[{'id': 'D3ADB33F1', 'name': 'general'}],
+            groups=[{'id': '1800IDGAF', 'name': 'super_secret_club'}],
+        )
+
+    slack_client = MagicMock(api_call=api_call)
+    return slack_client
+
+
+def test_layabout_can_register_handlers():
+    """
+    Test that layabout can register Slack API event handlers normally and as a
+    decorator, both with and without keyword arguments.
+    """
+    layabout = Layabout(name='test')
+    kwargs = dict(spam='🍳')
+
+    def handle_hello1(slack, event):
+        pass
+
+    @layabout.handle('hello')
```

Comment:
Oooo, OOO. I didn't realize you didn't do any value verification for event names.

So you can enter event names that are invalid and an error won't be thrown?

Reply:
Yeah, I purposefully didn't do any value verification for event names so I could be responsive to Slack updating their API without me updating this library.

New Slack RTM API event? Boom! Look at that. It's already supported out of the box from day one.

This does, of course, mean that you can enter invalid event names and an error won't be thrown, but there's little harm as you'll just never encounter such an event and the handler won't even be called.

Slack's own [`slackclient`](https://github.com/slackapi/python-slackclient) library doesn't validate their `SlackClient.api_call` method (presumably for the same reasons), so why should I?

Reply:
I have a strong personal preference towards having a enumerated set that reports when there's a version mismatch. I like being able to turn the check off but I find it very useful. 

I've just had too many experiences of screwing up an entire application by typing `http` instead of `https` or by registering `/var/messages` instead of `/var/log/messages`

I don't find the slack client not doing input validation a very convincing argument. Plenty of people use eventual consistency for use cases that require a single source of truth.

Does the slack client / API have a semantics version attached to it?

Reply:
The RTM API has [**way** too many methods](https://api.slack.com/rtm) for me to do that now and they change them often enough that I don't think I could responsibly keep up as a package maintainer.

As far as I can tell the RTM API does not have semantic versioning, or any such guarantees. The [Web API](https://api.slack.com/methods) does, but that won't really help here.

All I'm really saying is that I'm not going to be a good single source of truth for everyone to know what events are supported by the RTM API. They should look to Slack for that information,not this library. I can, of course, provide helpful links in the docs.